keps/sig-apps/85-Graduate-PDB-to-Stable/README.md

```diff
@@ -0,0 +1,435 @@
+type PodDisruptionBudgetList struct {
+    metav1.TypeMeta
+    // +optional
+    metav1.ListMeta
+    Items []PodDisruptionBudget
+}
+
+// PodDisruptionBudgetSpec is a description of a PodDisruptionBudget.
+type PodDisruptionBudgetSpec struct {
+    // An eviction is allowed if at least "minAvailable" pods selected by
+    // "selector" will still be available after the eviction, i.e. even in the
+    // absence of the evicted pod.  So for example you can prevent all voluntary
+    // evictions by specifying "100%".
+    // +optional
+    MinAvailable *intstr.IntOrString
+
+    // Label query over pods whose evictions are managed by the disruption
+    // budget.
+    // +optional
+    Selector *metav1.LabelSelector
+
+    // An eviction is allowed if at most "maxUnavailable" pods selected by
+    // "selector" are unavailable after the eviction, i.e. even in absence of
+    // the evicted pod. For example, one can prevent all voluntary evictions
+    // by specifying 0. This is a mutually exclusive setting with "minAvailable".
+    // +optional
+    MaxUnavailable *intstr.IntOrString
+}
+
+// PodDisruptionBudgetStatus represents information about the status of a
+// PodDisruptionBudget. Status may trail the actual state of a system.
+type PodDisruptionBudgetStatus struct {
+    // Most recent generation observed when updating this PDB status. DisruptionsAllowed and other
+    // status information is valid only if observedGeneration equals to PDB's object generation.
+    // +optional
+    ObservedGeneration int64
+
+    // DisruptedPods contains information about pods whose eviction was
+    // processed by the API server eviction subresource handler but has not
+    // yet been observed by the PodDisruptionBudget controller.
+    // A pod will be in this map from the time when the API server processed the
+    // eviction request to the time when the pod is seen by PDB controller
+    // as having been marked for deletion (or after a timeout). The key in the map is the name of the pod
+    // and the value is the time when the API server processed the eviction request. If
+    // the deletion didn't occur and a pod is still there it will be removed from
+    // the list automatically by PodDisruptionBudget controller after some time.
+    // If everything goes smooth this map should be empty for the most of the time.
+    // Large number of entries in the map may indicate problems with pod deletions.
+    // +optional
+    DisruptedPods map[string]metav1.Time
+
+    // Number of pod disruptions that are currently allowed.
+    DisruptionsAllowed int32
+
+    // current number of healthy pods
+    CurrentHealthy int32
+
+    // minimum desired number of healthy pods
+    DesiredHealthy int32
+
+    // total number of pods counted by this disruption budget
+    ExpectedPods int32
+
+    // Represents the latest available observations of a pdb's current state.
+    Conditions []DeploymentCondition
```

Comment:
you should use metav1.Condition instead (we're trying to unify conditions)

https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L1368

Reply:
Fixed